wlauto/common/android/device.py

```diff
@@ -109,6 +109,15 @@ def is_rooted(self):
     def abi(self):
         return self.getprop()['ro.product.cpu.abi'].split('-')[0]
 
+    """
+    Returns the ABI for this device.
+    This is the same as the ``abi`` property except it does not rely on the ``_is_ready`` flag
+    being set, and thus can be run directly from the host console. Useful when we need to provide
+    ABI-specific handling during initialization before the property is set.
+    """
+    def get_abi(self):
+        return adb_shell(self.adb_name, 'getprop ro.product.cpu.abi').split('-')[0]
+
```

Comment:
You should be using abi() above. The ``_is_ready`` flag is set once a connection to the device has been established; You cannot assume you're able to execute arbitrary commands on the device before the flag is set.